src/eos-updater-poll.c

```diff
@@ -203,6 +203,192 @@ get_fetchers (SourcesConfig *config,
+   * check again. */
+  do
+    {
+      const OstreeCollectionRef *refs[] = { NULL, NULL };
+      g_autoptr(GVariant) pull_options = NULL;
+      g_autoptr(GAsyncResult) find_result = NULL, pull_result = NULL;
+      g_auto(GVariantBuilder) builder = G_VARIANT_BUILDER_INIT (G_VARIANT_TYPE ("a{sv}"));
+
+      refs[0] = upgrade_collection_ref;
+      g_debug ("%s: Finding remotes advertising upgrade_collection_ref: (%s, %s)",
+               upgrade_collection_ref->collection_id, upgrade_collection_ref->ref_name);
+
+      ostree_repo_find_remotes_async (repo, refs, NULL  /* options */,
+                                      (OstreeRepoFinder **) finders->pdata,
+                                      NULL  /* progress */,
+                                      cancellable, async_result_cb, &find_result);
+
+      while (find_result == NULL)
+        g_main_context_iteration (context, TRUE);
+
+      results = ostree_repo_find_remotes_finish (repo, find_result, error);
+      if (results == NULL)
+        return NULL;
+
+      /* No updates available. */
+      if (results[0] == NULL)
+        {
+          g_message ("Poll: Couldn’t find any updates");
+          return NULL;
+        }
+
+      g_variant_builder_add (&builder, "{s@v}", "flags",
+                             g_variant_new_variant (g_variant_new_int32 (OSTREE_REPO_PULL_FLAGS_COMMIT_ONLY)));
+      pull_options = g_variant_ref_sink (g_variant_builder_end (&builder));
+
+      ostree_repo_pull_from_remotes_async (repo,
+                                           (const OstreeRepoFinderResult * const *) results,
+                                           pull_options, NULL  /* progress */, cancellable,
+                                           async_result_cb, &pull_result);
+
+      while (pull_result == NULL)
+        g_main_context_iteration (context, TRUE);
+
+      if (!ostree_repo_pull_from_remotes_finish (repo, pull_result, error))
+        return NULL;
+
+      g_clear_pointer (&new_refspec, g_free);
+      g_clear_pointer (&new_collection_ref, ostree_collection_ref_free);
+
+      /* Parse the commit and check there’s no redirection to a new ref. */
+      if (!parse_latest_commit (repo, upgrade_refspec, &checksum, &new_refspec,
+                                NULL, cancellable, error))
+        return NULL;
+
+      if (new_refspec != NULL)
+        upgrade_refspec = new_refspec;
+      if (new_collection_ref != NULL)
+        upgrade_collection_ref = new_collection_ref;
+    }
+  while (checksum == NULL);
+
+  /* Final checks on the commit we found. */
+  if (!is_checksum_an_update (repo, checksum, &commit, error))
+    return NULL;
+
+  if (commit == NULL)
+    return NULL;
+
+  info = eos_update_info_new (checksum, commit,
+                              upgrade_refspec, booted_refspec,
+                              NULL, g_steal_pointer (&results));
+  metrics_report_successful_poll (info);
```

Comment:
I think this function is not defined in this commit (it is in the next one)

Reply:
Oops, I didn’t check compilability between commits after rebasing and squashing.